plugins/profiles.koplugin/main.lua

```diff
@@ -113,22 +112,44 @@ function Profiles:getSubMenuItems()
                 end,
                 separator = true,
             },
+            {
+                text = _("Show in action list"),
+                checked_func = function()
+                    return v.settings.registered
+                end,
+                callback = function(touchmenu_instance)
+                    if v.settings.registered then
+                        dispatcherRemoveProfile(k)
+                        self.data[k].settings.registered = nil
+                    else
+                        dispatcherRegisterProfile(k)
+                        self.data[k].settings.registered = true
+                    end
+                    self.updated = true
+                    local actions_sub_menu = {}
+                    Dispatcher:addSubMenu(self, actions_sub_menu, self.data, k)
+                    touchmenu_instance.item_table[4].sub_item_table = actions_sub_menu -- item index in submenu
```

Comment:
does `touchmenu_instance:updateItems()` instead of manually regenerating the sub_menu?

Reply:
`getSubMenuItems()` + `updateItems()` switches the menu to the upper level.